WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -560,7 +573,22 @@ private void LogCoinJoinSummary(ImmutableArray<AliceClient> registeredAliceClien
 			consolidationMode ? MaxInputsRegistrableByWallet : GetInputTarget(allowedNonPrivateCoins.Count, privateCoins.Length, rnd));
 
 		// Let's allow only a inputCount - 1 private coins to play.
-		var allowedPrivateCoins = AnonScoreBiasedShuffle(privateCoins).Take(inputCount - 1);
+		var biasSuffledPrivateCoins = AnonScoreBiasedShuffle(privateCoins).ToArray();
+		IEnumerable<SmartCoin> allowedPrivateCoins;
+		lock (SecondLargestDenomOutputLock)
+		{
+			if (SecondLargestDenomOutput is null)
+			{
+				allowedPrivateCoins = biasSuffledPrivateCoins.Take(inputCount - 1);
+			}
+			else
+			{
+				// Deprioritize private coins those are too large.
+				var smallerPrivateCoins = biasSuffledPrivateCoins.Where(x => x.Amount <= SecondLargestDenomOutput);
+				var largerPrivateCoins = biasSuffledPrivateCoins.Where(x => x.Amount > SecondLargestDenomOutput);
+				allowedPrivateCoins = smallerPrivateCoins.Concat(largerPrivateCoins).Take(inputCount - 1);
```

Comment:
Do we want to prioritize small coins? Take will get from the beginning.

Reply:
Yes, that is the goal of this PR. Note: prioritize private coins those are under `SecondLargestDenomOutput`. Compare the master to this PR:

Assume you're a whale and you coinjoined your entire wallet out. Then you receive a new coin. AS of new coin will of course be 1. Therefore you'll have to coinjoin again. How do you select the inputs?

On master with a decent probability, on this PR with a low probability, you'll select a large private coin to accompany your new coin. Since large outputs' AS gets punished to the lowest input's AS, we have a problem, because your wallet privacy percentage will jump back to 70% from 99% after the first coinjoin.

This is the problem this PR is attempting to solve.